File: apps/demo/src/app/_models/maze/mock-mazes.ts
import { IMaze, IMovement, INewMaze } from './maze';

const graph1 = [
  ['S', 'O', 'X', 'X', 'X', 'X', 'X', 'X', 'X', 'X'],
  ['O', 'O', 'O', 'X', 'X', 'X', 'X', 'X', 'X', 'X'],
  ['O', 'X', 'O', 'O', 'O', 'X', 'O', 'O', 'O', 'O'],
  ['X', 'X', 'X', 'X', 'O', 'X', 'O', 'X', 'X', 'O'],
  ['O', 'O', 'O', 'O', 'O', 'O', 'O', 'X', 'X', 'O'],
  ['O', 'X', 'X', 'O', 'X', 'X', 'X', 'X', 'X', 'O'],
  ['O', 'O', 'O', 'O', 'X', 'X', 'X', 'X', 'X', 'E'],
];

const graph2 = [
  ['S', 'O', 'X', 'X', 'X', 'X', 'X', 'X', 'X', 'X'],
  ['O', 'O', 'O', 'X', 'X', 'X', 'X', 'X', 'X', 'X'],
  ['O', 'X', 'O', 'O', 'O', 'X', 'O', 'O', 'O', 'O'],
  ['X', 'X', 'X', 'X', 'O', 'X', 'O', 'X', 'X', 'O'],
  ['O', 'O', 'O', 'O', 'O', 'O', 'O', 'X', 'X', 'O'],
  ['O', 'X', 'X', 'X', 'X', 'X', 'O', 'X', 'X', 'X'],
  ['X', 'X', 'E', 'O', 'O', 'O', 'O', 'X', 'X', 'X'],
];

export const MOCK_MAZES: IMaze[] = [
  {
    id: 12,
    uploadDate: new Date(2022, 7, 18, 15, 20, 12),
    // graphString: JSON.stringify(graph1),
    // graphString: 'SOXXXXXXXX#OOOXXXXXXX#OXOOOXOOOO#XXXXOXOXXO#OOOOOOOXXO#OXXOXXXXXO#OOOOXXXXXE#',
    graph: graph1,
    start: { row: 0, col: 0 },
    exit: { row: 6, col: 9 },
  },
  {
    id: 13,
    uploadDate: new Date(2022, 7, 18, 17, 23, 42),
    // graphString: JSON.stringify(graph2),
    // graphString: 'SOXXXXXXXX#OOOXXXXXXX#OXOOOXOOOO#XXXXOXOXXO#OOOOOOOXXO#OXXXXXOXXX#XXEOOOOXXX#',
    graph: graph2,
    start: { row: 0, col: 0 },
    exit: { row: 6, col: 2 },
  },
];

export const NEW_MAZE_REQ: INewMaze = {
  graphString: 'SOXXXXXXXX#OOOXXXXXXX#OXOOOXOOOO#XXXXOXOXXO#OOOOOOOXXO#OXXOXXXXXO#OOOOXXXXXE#',
  start: { row: 0, col: 0 },
  exit: { row: 6, col: 9 },
};

export const VALID_MOVEMENTS: IMovement[] = [
  { name: 'Up', direction: { row: -1, col: 0 } },
  { name: 'Down', direction: { row: 1, col: 0 } },
  { name: 'Left', direction: { row: 0, col: -1 } },
  { name: 'Right', direction: { row: 0, col: 1 } },
];

File: ValantDemoApi/ValantDemoApi/MockData/MockMazes.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using ValantDemoApi.ValantMaze;
using ValantDemoApi.ValantMaze.Models;

namespace ValantDemoApi.MockData
{
  public class MockMazes
  {
    public MockMazes()
    {
      DateTime dt1 = new(2022, 8, 18, 13, 15, 22);
      TestMaze1 = new Maze
      {
        Id = 12,
        UploadDate = dt1.ToUniversalTime(),
        GraphString = "SOXXXXXXXX#OOOXXXXXXX#OXOOOXOOOO#XXXXOXOXXO#OOOOOOOXXO#OXXOXXXXXO#OOOOXXXXXE#",
        StartRow = 0,
        StartCol = 0,
        ExitRow = 6,
        ExitCol = 9
      };

      DateTime dt2 = new(2022, 8, 18, 16, 29, 52);
      TestMaze2 = new Maze
      {
        Id = 13,
        UploadDate = dt2.ToUniversalTime(),
        GraphString = "SOXXXXXXXX#OOOXXXXXXX#OXOOOXOOOO#XXXXOXOXXO#OOOOOOOXXO#OXXXXXOXXX#XXEOOOOXXX#",
        StartRow = 0,
        StartCol = 0,
        ExitRow = 6,
        ExitCol = 2
      };

      NewMazeRequest = new PostNewMazeDto
      {
        GraphString = "XOXXXXXXXX#OOOXOOOOEX#OXOOOXXOOO#XXOXOXOXXO#OOOOOOOXXO#OXXOXXSXXO#OOOOXXXXXX#",
        Start = new Cell(5, 6),
        Exit = new Cell(1, 8)
      };

    }

    public Maze TestMaze1 { get; set; }
    public Maze TestMaze2 { get; set; }
    public PostNewMazeDto NewMazeRequest { get; set; }
  }
}

